deepdrivemd/deepdrivemd_stream.py

```diff
@@ -0,0 +1,246 @@
+from deepdrivemd.utils import parse_args
+
+
+def generate_task(cfg: BaseStageConfig) -> Task:
+    task = Task()
+    task.cpu_reqs = cfg.cpu_reqs.dict().copy()
+    task.gpu_reqs = cfg.gpu_reqs.dict().copy()
+    task.pre_exec = cfg.pre_exec.copy()
+    task.executable = cfg.executable
+    task.arguments = cfg.arguments.copy()
+    return task
+
+
+class PipelineManager:
+    MOLECULAR_DYNAMICS_STAGE_NAME = "MolecularDynamics"
+    AGGREGATION_STAGE_NAME = "Aggregating"
+    MACHINE_LEARNING_STAGE_NAME = "MachineLearning"
+    AGENT_STAGE_NAME = "Agent"
+
+    MOLECULAR_DYNAMICS_PIPELINE_NAME = "MolecularDynamicsPipeline"
+    AGGREGATION_PIPELINE_NAME = "AggregatingPipeline"
+    MACHINE_LEARNING_PIPELINE_NAME = "MachineLearningPipeline"
+    AGENT_PIPELINE_NAME = "AgentPipeline"
+
+    def __init__(self, cfg: StreamingExperimentConfig):
+        self.cfg = cfg
+        self.stage_idx = 0
+        self.api = DeepDriveMD_API(cfg.experiment_directory)
+
+        self.pipelines = {}
+
+        p_md = Pipeline()
+        p_md.name = self.MOLECULAR_DYNAMICS_PIPELINE_NAME
+
+        self.pipelines[p_md.name] = p_md
+
+        p_aggregate = Pipeline()
+        p_aggregate.name = self.AGGREGATION_PIPELINE_NAME
+
+        self.pipelines[p_aggregate.name] = p_aggregate
+
+        p_ml = Pipeline()
+        p_ml.name = self.MACHINE_LEARNING_PIPELINE_NAME
+        self.pipelines[p_ml.name] = p_ml
+
+        p_outliers = Pipeline()
+        p_outliers.name = self.AGENT_PIPELINE_NAME
+        self.pipelines[p_outliers.name] = p_outliers
+
+        self._init_experiment_dir()
+
+    def _init_experiment_dir(self):
+        # Make experiment directories
+        self.cfg.experiment_directory.mkdir()
+        self.api.molecular_dynamics_stage.runs_dir.mkdir()
+        self.api.aggregation_stage.runs_dir.mkdir()
+        self.api.machine_learning_stage.runs_dir.mkdir()
+        self.api.agent_stage.runs_dir.mkdir()
+
+    def _generate_pipeline_iteration(self):
+
+        self.pipelines[self.MOLECULAR_DYNAMICS_PIPELINE_NAME].add_stages(
+            self.generate_molecular_dynamics_stage()
+        )
+        self.pipelines[self.AGGREGATION_PIPELINE_NAME].add_stages(
+            self.generate_aggregating_stage()
+        )
+        self.pipelines[self.MACHINE_LEARNING_PIPELINE_NAME].add_stages(
+            self.generate_machine_learning_stage()
+        )
+        self.pipelines[self.AGENT_PIPELINE_NAME].add_stages(self.generate_agent_stage())
+
+        self.stage_idx += 1
+
+    def generate_pipelines(self) -> List[Pipeline]:
+        self._generate_pipeline_iteration()
+        return self.pipelines.values()
+
+    def generate_molecular_dynamics_stage(self) -> Stage:
+        stage = Stage()
+        stage.name = self.MOLECULAR_DYNAMICS_STAGE_NAME
+        cfg = self.cfg.molecular_dynamics_stage
+        stage_api = self.api.molecular_dynamics_stage
+
+        init_file = glob.glob(str(cfg.task_config.initial_pdb_dir) + "/*.pdb")[0]
```

Comment:
Can use pathlib glob function and remove glob import. 
`init_file = str(list(cfg.task_config.initial_pdb_dir.glob("/*.pdb"))[0])`

This will be updated in the future to pass a different PDB file for each task. That will make it slightly cleaner:
`pdb_files = cfg.task_config.initial_pdb_dir.glob("/*.pdb")`

Reply:
This is actually not used anymore. Different pdb files for different tasks is already implemented in init_input() of run_openmm.py

Reply:
Thanks for removing it.